### PR TITLE
Added authentication and authorization docs for AAD.

### PR DESCRIPTION
**Pull request by jimmyca15**
This PR adds protocol documentation for AAD authentication and authorization. It also establishes a new pattern for documenting authentication vs authorization. This is handled by having a separate authentication and authorization folder. Authentication links have been updated to point to either the authentication index page or the relevant authentication scheme page.

The content of this doc set has already been reviewed internally, but feel free to give another go at reviewing the content.

### CODE REVIEW

File: docs/REST/authentication/index.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿# Authentication
+
+All HTTP requests must be authenticated. The following authentication schemes are supported.
+
+## HMAC
+
+[HMAC authentication](./authentication/hmac.md) uses a randomly generated secret to sign request payloads. Details on how requests using this authentication method are authorized can be found in the [HMAC authorization](../authorization/hmac.md) section.
```
Comment (et13) [inline docs/REST/authentication/index.md]: >./authentication/hmac.md [](start = 22, length = 24)

This link doesn't work - I'm seeing it resolve to authentication/authentication/hmac.md

File: docs/REST/authentication/index.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+## Azure Active Directory
+
+[Azure Active Directory (AAD) authentication](./authentication/aad.md) utilizes a bearer token that is obtained from Azure Active Directory to authenticate requests. Details on how requests using this authentication method are authorized can be found in the [AAD authorization](../authorization/aad.md) section.
```
Comment (et13) [inline docs/REST/authentication/index.md]: >./authentication/aad.md [](start = 46, length = 23)

This link doesn't work - I'm seeing it resolve to authentication/authentication/aad.md

File: docs/REST/authorization/hmac.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# HMAC Authorization - REST API Reference
+
+When HMAC authentication is used, operations fall in to one of two categories, read or write. Read-write access keys grant permission to call all operations. Read-only access keys grant permission to call only read operations. Whether an access key is read only or read-write is determined by its `readOnly` property. Any attempt to make a write request with a read only access key will result in the request being unauthorized.
```
Comment (et13) [inline docs/REST/authorization/hmac.md]: >read only [](start = 252, length = 9)

read-only?  (to match earlier usage)

Comment (et13) [inline docs/REST/authorization/hmac.md]: >read only [](start = 361, length = 9)

read-only?  (to match earlier usage)

File: docs/REST/authentication/index.md (hunk 1 above)
Comment (jimmyca15) [reply docs/REST/authentication/index.md]: Yes it was broken. Just fixed it, thanks!

File: docs/REST/authentication/index.md (hunk 2 above)
Comment (jimmyca15) [reply docs/REST/authentication/index.md]: Yes it was broken. Just fixed it, thanks!

File: docs/REST/authorization/hmac.md (hunk 3 above)
Comment (jimmyca15) [reply docs/REST/authorization/hmac.md]: Good catch, updated.

Comment (jimmyca15) [reply docs/REST/authorization/hmac.md]: Good catch, updated.

**Pull request merged**